Add Visitors to Brush ASTs

timraymond opened the pull request:
Certain applications would benefit greatly from being able to use
Visitors to traverse a Brush AST beforehand. This adds a Visitor
interface which requires Accept methods for the three major leaf nodes
of a Brush AST.

timraymond committed:
Add Visitors to Brush ASTs

Certain applications would benefit greatly from being able to use
Visitors to traverse a Brush AST beforehand. This adds a Visitor
interface which requires Accept methods for the three major leaf nodes
of a Brush AST.

lukebergen commented on parse/nodes.go:
```diff
@@ -10,6 +10,17 @@ import (
 // Node in a Braai AST.
 type Node interface {
 	Execute(*HandlerMux) (string, error)
```
Formatting weirdness?

lukebergen commented:
Looks good to me. Nice.

timraymond committed:
Add a CompositeVisitor

This Visitor will take multiple Visitors and invoke their corresponding
Accept methods as they are invoked on the CompositeVisitor. This allows
for one traversal of a Brush AST with multiple Visitors.
